src/main/java/net/ripe/ipresource/IpResourceSet.java

```diff
@@ -198,7 +198,7 @@ public void retainAll(IpResourceSet other) {
         while (thisResource != null && thatResource != null) {
             IpResource intersect = thisResource.intersect(thatResource);
```

Comment:
I don't know this codebase, but can you elaborate on why the normalisation happens here instead of in intersect (javadoc on `intersect`s post-conditions would have helped)?

Reply:
Tough one...